lib/IRGen/GenStruct.cpp

```diff
@@ -61,6 +63,47 @@ static StructTypeInfoKind getStructTypeInfoKind(const TypeInfo &type) {
   return (StructTypeInfoKind) type.getSubclassKind();
 }
 
+clang::CXXDestructorDecl *swift::irgen::getDestructor(SILType type) {
+  DestructorDecl *destructor = nullptr;
+  if (auto structDecl = type.getStructOrBoundGenericStruct()) {
+    for (auto member : structDecl->getMembers()) {
```

Comment:
See corresponding comment here:

https://github.com/apple/swift/pull/32378#discussion_r439954513

I would suggest iterating over the members of the C++ class instead of the members of the Swift struct. I believe this also avoids having to add a deinitializer to the Swift struct (which I believe isn't really allowed, see above).